snappy_pipeline/workflows/tmb_calculation/__init__.py

```diff
@@ -0,0 +1,228 @@
+from collections import OrderedDict
+import os
+import sys
+
+from biomedsheets.shortcuts import CancerCaseSheet, CancerCaseSheetOptions, is_not_background
+from snakemake.io import expand
+
+from snappy_pipeline.utils import dictify, listify
+from snappy_pipeline.workflows.abstract import BaseStep, BaseStepPart, LinkOutStepPart
+from snappy_pipeline.workflows.ngs_mapping import NgsMappingWorkflow, ResourceUsage
+from snappy_pipeline.workflows.somatic_variant_calling import (
+    SOMATIC_VARIANT_CALLERS_MATCHED,
+    SomaticVariantCallingWorkflow,
+)
+
+#: Extensions of files to create as main payload
+EXT_VALUES = (".json",".json.md5")
+
+#: Names of the files to create for the extension
+EXT_NAMES = ("json", "json_md5")
+
+#: Default configuration for the tmb calculation step
+DEFAULT_CONFIG = r"""
+step_config:
+    tmb_calculation:
+        path_somatic_variant_calling: ../somatic_variant_calling   # REQUIRED
+        tools_ngs_mapping: []      # default to those configured for ngs_mapping
+        tools_somatic_variant_calling: []  # default to those configured for somatic_variant_calling
+"""
```

Comment:
The user must have control on the bed file for the regions considered for TMB. It must be part of the configuration.